pygmo/_py_algorithms.py

```diff
@@ -0,0 +1,131 @@
+import random
+
+try:
+    from scipy.optimize import minimize
+
+    class scipy:
+        """
+        This class is a user defined algorithm (UDA) providing a wrapper around the function scipy.optimize.minimize.
+        The constructor accepts those arguments that are specific to the algorithm:
+        - args
+        - method
+        - tol - the tolerance
+        - callback
+        - options
+
+        Other aspects, like bounds or the existence of a gradient and hessian, are taken from the problem.
+        """
+
+        def __init__(self, args=(), method=None, tol=None, callback=None, options=None):
+            method_list = [
+                "Nelder-Mead",
+                "Powell",
+                "CG",
+                "BFGS",
+                "Newton-CG",
+                "L-BFGS-B",
+                "TNC",
+                "COBYLA",
+                "SLSQP",
+                "trust-constr",
+                "dogleg",
+                "trust-ncg",
+                "trust-exact",
+                "trust-krylov",
+            ]
+            if method in method_list + [None]:
+                self.method = method
+            else:
+                raise ValueError("Method not supported: " + method)
+
+            self.args = args
+            self.tol = tol
+            self.callback = callback
+            self.options = options
+
+        def evolve(self, population):
+            """
+            Take a random member of the population, use it as initial guess
```

Comment:
In pagmo we have a class "https://github.com/esa/pagmo2/blob/master/include/pagmo/algorithms/not_population_based.hpp"
that deals with non population based algorithms. I think we should mirror that set up here. This is the first python pure UDA we are adding, thus its important we do it right ...

Could you also add that class in python and make this UDA inherit from it? @bluescarni Do you agree this is the way to go?

Reply:
> Could you also add that class in python and make this UDA inherit from it? @bluescarni Do you agree this is the way to go?

The ``not_population_based`` interface is a bit redundant now because we have the individual selection policies from the topology machinery that we could re-use here. What about embedding an ``s_policy`` object in the UDA with a resonable default (i.e., replace best)?

Reply:
Since 77f8218, the wrapper accepts an s_policy object in the constructor.